sphinx_immaterial/dropdown.py

```diff
@@ -0,0 +1,78 @@
+"""Taken from sphinxcontrib-details
+"""
+from docutils import nodes
```

Comment:
It seems like this is effectively just a different way of styling an admonition.  I wonder if it would make sense to just augment the built-in admonition directives with an additional option to indicate if they should behave as `details` elements --- and possibly that could be upstreamed in sphinx itself.

Reply:
I did look into that. It needs to be a `container` node to customize the HTML tag name. It only behaves like a admonition in this theme because of the CSS rule inherited from mkdocs-material.

Reply:
If we get rid of the CSS rule, the dropdown would not be styled like an admonition because there is nothing in dropdown.py that is meant to trigger the CSS admonition styles.

UPDATE: I made the dropdown title optional, but had to add the `admonition` class manually when not using a `details` tag. This could be done better if/when we exclude the inherited CSS from mkdocs-material about `details`/`summary` tags.
https://github.com/jbms/sphinx-immaterial/blob/ec14e7cda42f4bd314edfe29294ade3ab0ccc47d/src/assets/stylesheets/main/extensions/pymdownx/_details.scss#L38-L39

https://github.com/jbms/sphinx-immaterial/blob/ec14e7cda42f4bd314edfe29294ade3ab0ccc47d/src/assets/stylesheets/main/extensions/pymdownx/_details.scss#L72-L73

Reply:
I'm happy to remove the include of `main/extensions/pymdownx/_details` if you think that is the best solution.

Reply:
It would introduce a bit more testing, but it would also make maintenance easier as we wouldn't need our own directive.

I do wonder how we would support admonitions without a title at that point because the rst directive requires 1 arg (a title), and the arg might be limited to 1 line of input...